crates/ruff_python_formatter/src/statement/stmt_class_def.rs

```diff
@@ -1,12 +1,113 @@
-use crate::{not_yet_implemented, FormatNodeRule, PyFormatter};
-use ruff_formatter::{write, Buffer, FormatResult};
-use rustpython_parser::ast::StmtClassDef;
+use crate::expression::parentheses::Parenthesize;
+use crate::prelude::*;
+use crate::trivia::{first_non_trivia_token, SimpleTokenizer, Token, TokenKind};
+use crate::USE_MAGIC_TRAILING_COMMA;
+use ruff_formatter::{format_args, write};
+use ruff_text_size::TextRange;
+use rustpython_parser::ast::{Expr, Keyword, Ranged, StmtClassDef};
 
 #[derive(Default)]
 pub struct FormatStmtClassDef;
 
 impl FormatNodeRule<StmtClassDef> for FormatStmtClassDef {
     fn fmt_fields(&self, item: &StmtClassDef, f: &mut PyFormatter) -> FormatResult<()> {
-        write!(f, [not_yet_implemented(item)])
+        let StmtClassDef {
+            range: _,
+            name,
+            bases,
+            keywords,
+            body,
+            decorator_list,
+        } = item;
+
+        f.join_with(hard_line_break())
+            .entries(decorator_list.iter().formatted())
+            .finish()?;
+
+        if !decorator_list.is_empty() {
+            hard_line_break().fmt(f)?;
+        }
+
+        write!(f, [text("class"), space(), name.format()])?;
+
+        if !(bases.is_empty() && keywords.is_empty()) {
+            write!(
+                f,
+                [group(&format_args![
+                    text("("),
+                    soft_block_indent(&FormatInheritanceClause {
+                        class_definition: item
+                    }),
+                    text(")")
+                ])]
+            )?;
+        }
+
+        write!(f, [text(":"), block_indent(&body.format())])
+    }
+}
+
+struct FormatInheritanceClause<'a> {
+    class_definition: &'a StmtClassDef,
+}
+
+impl Format<PyFormatContext<'_>> for FormatInheritanceClause<'_> {
+    fn fmt(&self, f: &mut Formatter<PyFormatContext<'_>>) -> FormatResult<()> {
+        let StmtClassDef {
+            bases,
+            keywords,
+            name,
+            ..
+        } = self.class_definition;
+
+        let separator = format_with(|f| write!(f, [text(","), soft_line_break_or_space()]));
```

Comment:
We should probably extract a helper at some point that joins a list with `,` and `soft_line_break_or_space` and preserves any trailing magic comma.